Documentation/MixedRealityConfigurationGuide.md

```diff
@@ -359,23 +353,99 @@ using UnityEngine;
-        MixedRealityToolkit.Instance.ActiveProfile = profile;
+        MixedRealityToolkit.SetProfileBeforeInitialization(ProfileToUse);
     }
 }
 ```
 
-Instead of "RuntimeSwapparoo.asset", it's possible to have some arbitrary set of profiles which apply to
+Instead of "ProfileToUse", it's possible to have some arbitrary set of profiles which apply to
 specific platforms (for example, one for HoloLens 1, one for VR, one for HoloLens 2, etc). It's possible
-to use various other indicators (i.e. https://docs.unity3d.com/ScriptReference/SystemInfo.html, or
+to use various other indicators (e.g. https://docs.unity3d.com/ScriptReference/SystemInfo.html, or
 whether or not the camera is opaque/transparent), to figure out which profile to load.
 
+### Active profile switch
+
+This can be accomplished by setting the `MixedRealityToolkit.Instance.ActiveProfile` property to a new profile replacing the active profile.
+
+```csharp
+MixedRealityToolkit.Instance.ActiveProfile = ProfileToUse;
+```
+
+Note when setting `ActiveProfile` during runtime, the destroy of the currently running services will happen after the last LateUpdate() of all services, and the instantiation and initialization of the services associated with the new profile will happen before the first Update() of all services.
+
+A noticable application hesitation may occur during this process. Also any script with higher priority than the `MixedRealityToolkit` script can enter its Update before the new profile is properly setup. See [Script Execution Order settings](https://docs.unity3d.com/Manual/class-MonoManager.html) for more information on script priority.
+
+Special handling is required when Unity UI elements are present in the scene. Such elements are usually children of `Canvas` that often requires a reference to the world camera (UI camera). During an active profile switch, UI cameras are destroyed and recreated and thus the references to them have to be updated for all canvases with previously assigned world camera. `VerifyCanvasConfiguration` in the `CanvasUtility` script can be used to update these references. The following script is an example of one way to handle this problem.
+
+
+```csharp
+/// <summary>
+/// Sample MonoBehaviour that will change the active profile and update the references
+/// to UI camera for Canvases to ensure the functionality of Unity UI after the change.
+/// </summary>
+/// <remarks>
+/// Note that this script assumes there is no instantiation and/or destroy of Canvases.
+/// Canvases to be used with MRTK have the CanvasUtility script attached.
+/// </remarks>
+using Microsoft.MixedReality.Toolkit;
+using Microsoft.MixedReality.Toolkit.Input.Utilities;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ActiveProfileSwapper : MonoBehaviour
+{
+    
+    public MixedRealityToolkitConfigurationProfile ProfileToUse;
```

Comment:
We might want this to be private with the SerializeField tag, so the code snippets follow the MRTK coding guidelines